File: app/src/main/java/com/productivity/habits/domain/repository/GamificationRepository.kt
package com.productivity.habits.domain.repository

import com.productivity.habits.domain.gamification.AchievementStatus
import com.productivity.habits.domain.gamification.LevelUpCelebration
import com.productivity.habits.domain.gamification.PlayerProgression
import kotlinx.coroutines.flow.Flow

interface GamificationRepository {
    fun getPlayerProgression(): Flow<PlayerProgression>
    fun getAchievements(): Flow<List<AchievementStatus>>
    fun getPendingCelebration(): Flow<LevelUpCelebration?>
    suspend fun dismissCelebration(level: Int)
}

File: lib/domain/repositories/gamification_repository.dart
import '../gamification/gamification_models.dart';

abstract class GamificationRepository {
  Stream<PlayerProgression> getPlayerProgression();
  Stream<List<AchievementStatus>> getAchievements();
  Stream<LevelUpCelebration?> getPendingCelebration();
  Future<void> dismissCelebration(int level);
}
